[LiveComponent] Lazy load LiveComponent (#1515)
symfony/ux: src/LiveComponent/doc/index.rst

Deferring / Lazy Loading Components
-----------------------------------

> Defer Rendering

Reviewer: Before this header, let's add 1-2 sentences to quickly introduce defer vs lazy. I fear that the `defer` and `lazy` will be confusing, so let's very clearly give them the choice. Then they can read how to implement the one they want.

Author: I'll do it after we decide the "behaviour" / "naming", that will have an impact on the way to explain things. 

But it's noted !

Deferring / Lazy Loading Components
-----------------------------------

When a page loads, all components are rendered immediately. If a component is

> heavy to render, you can defer its rendering after the page has loaded.

Reviewer: ```suggestion
heavy to render, you can defer its rendering until after the page has loaded. This is done
by making an Ajax call to load the component's real content either as soon as the page loads
(``defer``) or when the component becomes visible (``lazy``).
```

Then I think we can remove the next 2 paragraphs. Shorter text is more likely to be read.

Author: Thank you, much better

Moreover, some components are not needed until the user scrolls to them (like
components at the bottom of a long page like comments, related products, etc.).
In this case, you can load them "lazily" when they are in the viewport.

By loading a component asynchronously, the page will load faster and the user
will be able to interact with it sooner, improving both the user experience
and the overall performance of your application.

.. note::

    Behind the scenes, your component *is* created & mounted during the initial

>     page load, but it isn't rendered. So keep your heavy work to methods in

Reviewer: ```suggestion
    page load, but its template isn't rendered. So keep your heavy work to methods in
```

>     your component (e.g. ``getProducts()``) that are only called when rendering.

Reviewer: ... called from the component's template.

> Loading "defer"

Reviewer: ```suggestion
Loading "defer" (Ajax on Load)
```

~~~~~~~~~~~~~~~

.. versionadded:: 2.13.0

    The ability to defer loading a component was added in Live Components 2.13.

If a component is heavy to render, you can defer rendering it until after
the page has loaded. To do this, add a ``loading="defer"`` attribute:

.. code-block:: html+twig

    {# With the component function #}
    <twig:SomeHeavyComponent loading="defer" />

.. code-block:: twig

    {# With the HTML syntax #}
    {{ component('SomeHeavyComponent', { loading: 'defer' }) }}

This renders an empty ``<div>`` tag, but triggers an Ajax call to render the
real component once the page has loaded.

> Loading "lazy"

Reviewer: ```suggestion
Loading "lazy" (Ajax when Visible)
```

~~~~~~~~~~~~~~

.. versionadded:: 2.17.0

    The ability to load a component "lazily" was added in Live Components 2.17.

The ``lazy`` option is similar to ``defer``, but it defers the loading of
the component until it's in the viewport. This is useful for components that
are far down the page and are not needed until the user scrolls to them.

To use this, set a ``loading="lazy"`` attribute to your component:

.. code-block:: html+twig

    {# With the HTML syntax #}
    <twig:Acme foo="bar" loading="lazy" />

.. code-block:: twig

    {# With the Twig syntax #}
    {{ component('SomeHeavyComponent', { loading: 'lazy' }) }}

This renders an empty ``<div>`` tag. The real component is only rendered when
it appears in the viewport.

> Defer or Lazy ?

Reviewer: ```suggestion
Defer or Lazy?
```

I know, that's a style thing that varies by language, but without the space is the way in the Symfony docs.

Author: 100%, just a french habit :)
